Publish minutes of 2025-09-25 meeting (#874)
hober/web-extensions: _minutes/2025-09-25-wecg.md

 * [rob] Let's wait for input from Google's tech lead on this, and then we can see how we can converge.

[PR 873](https://github.com/w3c/webextensions/pull/873): Add specification for `externally_connectable`

 * [kiara] Early draft, to request feedback. Question is how we document APIs; should we use separate files or one?
 * [simeon] Currently one file, for simplicity we can continue that. Eventually we will probably split it. I have an agenda item for the WG discussion on structuring deliverables.
 * [tomislav] CSS spec was split in order to deliver specific modules separately; I think that we can take the living standards approach like HTML.
 * [timothy] I'm for a single doc, at least initially, until it gets unwieldy.
 * [simeon] What I saw in specs is organization by capabilities, and sections with descriptions of operations. Separate from IDL descriptions of the APIs. Were you thinking of manifest.json or also covering the APIs?
 * [kiara] Just the manifest.json piece.
 * [oliver] Happy to provide an initial review from the Chrome side.
 * [oliver] Out of curiosity, does Safari support externally_connectable?

>  * [kiara] We support externally_connectable, but not the ids key yet.

Reviewer: I don't know that Kiara said "yet" here. That seems more like the kind of thing I wouldn't expect from the Safari team.

```suggestion
 * [kiara] We support externally_connectable, but not the ids key.
```

 * [timothy] One of the complications is the ID passed on the website differs between Chrome and Safari. That has to be documented differently, but that can be part of the sendMessage spec.

>  * [tomislav] Is the value of the ID in Safari an object? String?.

Reviewer: ```suggestion
 * [tomislav] Is the value of the ID in Safari an object? String?
```

 * [timothy] A string, parentheses, spaces. An opaque string.
 * [mukul] Is there a length limit of the ID?
 * [timothy] No imposed limit, perhaps a limit on the bundle ID length; we don't have a limit. Typically 30-40 characters.

>  * [simeon] Firefox supports communications between extensions, not websites. Does Safari plan to support sendMessage between extensions.

Reviewer: ```suggestion
 * [simeon] Firefox supports communications between extensions, not websites. Does Safari plan to support sendMessage between extensions?
```

 * [oliver] Last time I looked into it on Firefox; someone can upload arbitrary add-ons to AMO to claim an ID. Is that still true?
 * [rob] Yes. Some ID formats are restricted, but generally any unused ID can be claimed.

[Issue 400](https://github.com/w3c/webextensions/issues/400): Inconsistency: runtime.getManifest() and missing keys

 * [carlos] Seems this issue has never been triaged. Seems the main problematic issue of the inconsistencies is the lack of chrome.runtime.getManifest() in content scripts in Safari.
 * [oliver] I recall discussing this on the Chrome side; the intention is returning the parsed version of the manifest, which can result in values being localized. We also used to return a current_locale key that is internal and should not be part of the API result but we do return it.
 * [rob] What about URLs? In Firefox we normalize the provided URLs.
 * [carlos] Chrome does not normalize relative URLs, resulting in issues when used in APIs such as action.setIcon.
 * [timothy] In Safari it is the same; other than localization, it is exactly what the developer puts there. We do not even prune keys.

>  * [simeon] In since case I've needed to use a combination of runtime.getManifest() and a fetch() of the manifest file to make sure I'm working with the data I expect.

Reviewer: ```suggestion
 * [simeon] In some cases I've needed to use a combination of runtime.getManifest() and a fetch() of the manifest file to make sure I'm working with the data I expect.
```

 * [timothy] I see, so it's like creating a script tag. I guess that would be a Firefox-specific behavior, though, as Chrome and Safari don't do that.
 * [rob] Possibly, yes. We may need to collect telemetry to gauge usage.
 * [oliver] My feeling is also that normalizing URLs feels very verbose. I'm inclined to leave it inconsistent until we see that it's causing issues.
 * [rob] … I think we can note in a spec that there's some level of normalization going on and use a non-normative note to provide some examples. At the top level we include unknown keys. That's the main question for this issue: how should we handle those?
 * [tomislav] Safari indicated they don't remove them. We (Firefox) leave top level keys but prune some sub-keys.
 * [timothy] Confirmed. Getting back to the use cases, if most people are just getting the version or manifest version number, I wonder how much it matters that normalization or removing unknown keys matters.

>  * [simeon] An example I can think of is browser_specific_settings.gecko.data_collection in Firefox; If a library wants to shim the behavior in older/other browser versions, they'd have to read the manifest.

Reviewer: ```suggestion
 * [simeon] An example I can think of is browser_specific_settings.gecko.data_collection_permissions in Firefox; if someone wanted to shim the behavior in older/other browser versions, they'd have to read the manifest.
```

 * [oliver] Would there be appetite if I were to provide a `runtime.getVersion()` method? Feels like a case where you shouldn't have to request the full manifest.
 * [simeon] Is there substantially more overhead from a browser engineering perspective for exposing the object vs. the number?
 * [rob] Most overhead is in having to send the manifest to all processes; after that there is little gain between returning the full manifest vs just the version.
 * [timothy] Yes. I suppose that it would only make a difference if we ever consider deprecating getManifest().
 * [rob] Which I don't see happening.

>  * [rob] If the use case is to return specific values, why wouldn't we expand `runtime.getManifest()` with a parameter that specifies the key the developer is interested in?

Reviewer: I remember that someone said the the intent here was to pave the cow paths on common patterns, but I don't recall who. @xeenon, maybe?

Author: Maybe @oliverdunk ?

Reviewer 2: I'm not sure when, but I definitely said something along those lines at one point. Feel free to drop it in if that helps the flow of the conversation :)
